src/hotspot/share/gc/shared/preservedMarks.hpp

```diff
@@ -57,6 +57,7 @@ class PreservedMarks {
 public:
   size_t size() const { return _stack.size(); }
   inline void push_if_necessary(oop obj, markWord m);
+  inline void push_always(oop obj, markWord m);
```

Comment:
I was thinking `push_always` makes a clearer contrast with its neighbor: the two are parallel and there's no interconnection/dependency btw them.

Reply:
Sure, but they do push elements the same way so it could be shared to avoid having to change two places if anything needs updating in the future. That said, the code is small and live close together so should be fine if you prefer it this way.

Reply:
I think it's better not to mix the two APIs. The duplication is very small after all.